pkg/tbtcpg/moving_funds.go

```diff
@@ -0,0 +1,628 @@
+			err,
+		)
+	}
+
+	currentBlockNumber, err := blockCounter.CurrentBlock()
+	if err != nil {
+		return nil, fmt.Errorf(
+			"failed to get current block number: [%w]",
+			err,
+		)
+	}
+
+	filterLookBackSeconds := uint64(MovingFundsCommitmentLookBackPeriod.Seconds())
+	filterLookBackBlocks := filterLookBackSeconds / uint64(mft.chain.AverageBlockTime().Seconds())
+	filterStartBlock := currentBlockNumber - filterLookBackBlocks
+
+	filter := &tbtc.MovingFundsCommitmentSubmittedEventFilter{
+		StartBlock:          filterStartBlock,
+		WalletPublicKeyHash: [][20]byte{sourceWalletPublicKeyHash},
+	}
+
+	events, err := mft.chain.PastMovingFundsCommitmentSubmittedEvents(filter)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"failed to get past moving funds commitment submitted events: [%w]",
+			err,
+		)
+	}
+
+	// Moving funds commitment can be submitted only once for a given wallet.
+	// Check just in case.
+	if len(events) != 1 {
+		return nil, fmt.Errorf(
+			"unexpected number of moving funds commitment submitted events: [%v]",
+			len(events),
+		)
+	}
+
+	targetWallets := events[0].TargetWallets
+
+	// Make sure all the target wallets are Live.
+	for _, targetWallet := range targetWallets {
+		targetWalletChainData, err := mft.chain.GetWallet(targetWallet)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"failed to get wallet data for target wallet [0x%x]: [%w]",
+				targetWallet,
+				err,
+			)
+		}
+
+		if targetWalletChainData.State != tbtc.StateLive {
+			return nil, fmt.Errorf(
+				"%w: [0x%x]",
+				ErrTargetWalletNotLive,
+				targetWallet,
+			)
+		}
+	}
+
+	// Just in case check if the hash of the target wallets matches the moving
+	// funds target wallets commitment hash.
+	packedWallets := []byte{}
+	for _, wallet := range targetWallets {
+		packedWallets = append(packedWallets, wallet[:]...)
+	}
+
+	calculatedHash := crypto.Keccak256(packedWallets)
+
+	if !bytes.Equal(calculatedHash, targetWalletsCommitmentHash[:]) {
+		return nil, ErrWrongCommitmentHash
+	}
+
+	return targetWallets, nil
+}
+
+// GetWalletMembersInfo returns the wallet member IDs based on the provided
+// wallet operator addresses. Additionally, it returns the position of the
+// moving funds task execution operator on the list.
+func (mft *MovingFundsTask) GetWalletMembersInfo(
+	walletOperators []chain.Address,
+	executingOperator chain.Address,
+) ([]uint32, uint32, error) {
+	walletMemberIDs := make([]uint32, 0)
+	walletMemberIndex := 0
+
+	for index, operatorAddress := range walletOperators {
+		operatorID, err := mft.chain.GetOperatorID(operatorAddress)
+		if err != nil {
+			return nil, 0, fmt.Errorf("failed to get operator ID: [%w]", err)
+		}
+
+		// Increment the index by 1 as operator indexing starts at 1, not 0.
+		// This ensures the operator's position is correctly identified in the
+		// range [1, walletOperators.length].
+		if operatorAddress == executingOperator {
+			walletMemberIndex = index + 1
+		}
```

Comment:
If `executingOperator` controls multiple members in the wallet, this will set `walletMemberIndex` to the last one. Not saying this is bad but, can we always use the first member? The coordinator does this for communication and using the same pattern here may facilitate potential debugging.

Also, let's move this check at the top of the loop to logically group related steps.

Reply:
> Just to make sure: when an operator controls multiple wallet members, then its address will appear multiple times on [the wallet's operator's list](https://github.com/keep-network/keep-core/blob/e5d8ec35ab2328bcb289d960512afcaf3b71f63f/pkg/tbtc/coordination.go#L202), right?

Yes!